From f007582c6646d0b0d8c46dae4ac82c7b5f23e8d7 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:43 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): setclientid, setclientid_confirm, renew

Message-id: <1410898849-22126-160-git-send-email-bfields@redhat.com>
Patchwork-id: 95827
O-Subject: [Patch RHEL7 159/165] nfsd: Remove nfs4_lock_state(): setclientid, setclientid_confirm, renew
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: cb86fb1428499160eea1be7612127952eec69122
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index e6b27ed..a4ae38b 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -2948,7 +2948,6 @@ nfsd4_setclientid(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if (new == NULL)
   return nfserr_jukebox;
  /* Cases below refer to rfc 3530 section 14.2.33: */
- nfs4_lock_state();
  spin_lock(&nn->client_lock);
  conf = find_confirmed_client_by_name(&clname, nn);
  if (conf) {
@@ -2983,7 +2982,6 @@ nfsd4_setclientid(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  status = nfs_ok;
 out:
  spin_unlock(&nn->client_lock);
- nfs4_unlock_state();
  if (new)
   free_client(new);
  if (unconf)
@@ -3006,7 +3004,6 @@ nfsd4_setclientid_confirm(struct svc_rqst *rqstp,
 
  if (STALE_CLIENTID(clid, nn))
   return nfserr_stale_clientid;
- nfs4_lock_state();
 
  spin_lock(&nn->client_lock);
  conf = find_confirmed_client(clid, false, nn);
@@ -3056,7 +3053,6 @@ out:
  spin_unlock(&nn->client_lock);
  if (old)
   expire_client(old);
- nfs4_unlock_state();
  return status;
 }
 
@@ -4113,7 +4109,6 @@ nfsd4_renew(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  __be32 status;
  struct nfsd_net *nn = net_generic(SVC_NET(rqstp), nfsd_net_id);
 
- nfs4_lock_state();
  dprintk("process_renew(%08x/%08x): starting\n", 
    clid->cl_boot, clid->cl_id);
  status = lookup_clientid(clid, cstate, nn);
@@ -4126,7 +4121,6 @@ nfsd4_renew(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
   goto out;
  status = nfs_ok;
 out:
- nfs4_unlock_state();
  return status;
 }
 
-- 
1.7.1